### ISSUE TITLE
Analyzer suggestion: Convert private field to constructor local variable

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#63803]
**Brief description:**

Reports fields that are only used inside a constructor.

**Languages applicable:**

Both C# and VB

**Code example that the analyzer should report:**

```csharp
class C
{
    private string _s;
    public C(string s)
    {
        _s = s ?? string.Empty;
        Console.WriteLine(_s);
    }
}
```

Codefix might produce:

```csharp
class C
{
    public C(string s)
    {
        // naming violation, would be awesome if the codefix can produce better code.
        string _s = s ?? string.Empty;
        Console.WriteLine(_s);
    }
}
```

**Additional information:**

I'm not sure whether this is more suitable as IDExxxx or CAxxxx.

**Documentation requirements:**

When this analyzer is implemented, it must be documented by following the steps at [Documentation for IDE CodeStyle analyzers](https://github.com/dotnet/roslyn/blob/main/docs/contributing/Documentation%20for%20IDE%20CodeStyle%20analyzers.md).

**Comment by CyrusNajmabadi** [comment]
Closingo ut due to lack of feedback on this.
